lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -876,6 +884,12 @@ def _start_opensearch(self, event: _StartOpenSearch) -> None:  # noqa: C901
             )
             self._post_start_init(event)
         except (OpenSearchHttpError, OpenSearchStartTimeoutError, OpenSearchNotFullyReadyError):
+            if "data" not in deployment_desc.config.roles:
```

Comment:
does this mean that we always start the opensearch daemon no matter the existence of a data node? 
Why can't we do it conditionally to the content of the roles of the `peer-cluster-fleet` in the peer-cluster-relation-joined event of the providers?

Reply:
has been updated.